lib/ain-grpc/src/rpc/eth.rs

```diff
@@ -357,7 +358,11 @@ impl MetachainRPCServer for MetachainRPCModule {
                 block_number,
             })
             .map_err(RPCError::EvmError)?;
-        Ok(Bytes(data))
+
+        match error_on_execution_failure(&exit_reason, &data) {
+            Ok(_) => Ok(Bytes(data)),
+            Err(e) => Err(e),
+        }
```

Comment:
Should consider passing the entire TxResponse instance into the error_on_execution_failure method, which returns the RpcResult<Bytes> itself to skip the entire match code block

Reply:
hmm.. not really sure what you mean.. but ExitReason match is still required